File: src/main/kotlin/com/practiceKotlin/lec15/Lec15Main.java
package com.practiceKotlin.lec15;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Lec15Main {

    public static void main(String[] args) {
        int[] array = {100, 200};

        for(int i=0; i<array.length; i++){
            System.out.printf("%s %s = ", i, array[i]);
        }
;
        final List<Integer> numbers = Arrays.asList(100, 200);

        //하나를 가져오기
        System.out.println(numbers.get(0));

        //for each
        for (int number : numbers) {
            System.out.println(number);
        }

        //전통적인 for문
        for (int i=0; i<numbers.size(); i++){
            System.out.printf("%s %s", i, numbers.get(i));
        }

        //JDK 8까지
        Map<Integer, String> oldMap = new HashMap<>();
        oldMap.put(1, "MONDAY");
        oldMap.put(2, "TUESDAY");

        //JDK 9부터
        Map.of(1, "Monday", 2, "TUESDAY");
    }
}

File: src/main/kotlin/com/practiceKotlin/lec15/lec15Main.kt
package com.practiceKotlin.lec15

import org.springframework.stereotype.Controller

fun main() {

    val array = arrayOf(100, 200)

    for(i in array.indices){
        println("${i} ${array[i]}")
    }

    for((idx, value) in array.withIndex()){
        println("$idx $value")
    }

    array.plus(300)

    val numbers = listOf(100, 200)
    val emptyList = emptyList<Int>()

    printNumbers(emptyList)

    println(numbers[0])

    for(number in numbers){
        println(number)
    }

    for((idx, value) in numbers.withIndex()){
        println("${idx} ${value}")
    }

    //가변 리스트 생성
    val numbers2 = mutableListOf(100, 200)
    numbers2.add(300)

    val oldMap = mutableMapOf<Int, String>()
    oldMap[1] = "MONDAY"
    oldMap[2] = "TUESDAY"

    mapOf(1 to "MONDAY", 2 to "TUESDAY")

    for(key in oldMap.keys){
        println(key)
        println(oldMap[key])
    }



}

private fun printNumbers(numbers: List<Int>){

}